test/metabase/db/data_migrations_test.clj

```diff
@@ -331,3 +334,100 @@
           (testing "And it is idempotent"
             (#'migrations/migrate-click-through)
             (is (= expected-settings (get-settings!)))))))))
+
+(defn- do-with-temporary-setting-value
+  "Temporarily update the value of a setting, execute `thunk`, then re-establish the original value"
+  [setting-k value thunk]
+  (assert (and (string? setting-k) (string? value))
+          (format "key and value should be string, got %s and %s" (type setting-k) (type value)))
+  (let [original-value (db/select-one-field :value Setting :key setting-k)]
+    (try
+     (if original-value
+       (db/update! Setting setting-k :value value)
+       (db/insert! Setting :key setting-k :value value))
+     (thunk)
+     (finally
+       (if original-value
+         (db/update! Setting setting-k :value original-value)
+         (db/delete! Setting :key setting-k))))))
```

Comment:
This looks like a less fully-featured copy of https://github.com/metabase/metabase/blob/5a67efea0e4812a75b699ff00cb68034c020d01e/test/metabase/test/util.clj#L345-L379

Reply:
It's better to extend the existing version to make it a little more flexible (e.g. to support raw values) so everyone can use that functionality instead of just copying it and stripping a bunch of stuff out and tweaking it.

Imagine if we did that every time we needed to do something slightly different than what existing test utils did -- we'd have a hundred different versions of every test util scattered all over the code base